message.go

```diff
@@ -51,15 +51,15 @@ func (m *Message) decodeToNull() error {
 	return err
 }
 
-func (m *Message) makeDecodeNext(debugHook func(interface{})) DecodeNext {
+func (m *message) makeDecodeNext(debugHook func(interface{})) DecodeNext {
 	// Reserve the next object
-	m.t.ReadLock()
+	m.t.Lock()
 	return func(i interface{}) error {
 		ret := m.Decode(i)
 		if debugHook != nil {
 			debugHook(i)
 		}
-		m.t.ReadUnlock()
```

Comment:
I didn't, it was always a `sync.Mutex`, it was just wrapped and named `ReadLock` and `ReadUnlock`. I'm planning on refactoring out this lock completely, but for now it's identical to the current version, just with less code.